src/extension.ts

```diff
@@ -23,11 +26,29 @@ const provideHover = async (document: vscode.TextDocument, position: vscode.Posi
   }
 
   const url = document.getText(range);
-  const res = await axios.get(url);
-  const meta = getMetaTags(res.data);
-  const hoverContent = formatHover(meta);
-  console.log(hoverContent.value);
-  return new vscode.Hover(hoverContent);
+
+  // Check if hover content for the URL is already cached
+  if (hoverContentCache.has(url)) {
+    const vsHoverContent = hoverContentCache.get(url)!;
+    return vsHoverContent;
+  }
+
+  try {
+    const response = await axios.get(url);
+    const meta = getMetaTags(response.data);
+    const hoverContent = formatHover(meta);
+
+    // Cache hover content for the URL
+    const vsHoverContent = new vscode.Hover(hoverContent);
+    hoverContentCache.set(url, vsHoverContent);
```

Comment:
Consider extracting all the cache mechanism to another file, with function like `getOrCreate(key, () => { /* generate new value */ })`

Reply:
I separated to a get and set functions.
a getOrCreate will require to move the whole axios get process there too.